hazardous/metrics/_concordance_index.py

```diff
@@ -0,0 +1,409 @@
+        on features (for instance Cox). Not used if ipcw=None or 'km'.
+
+    event_of_interest: int, default=1
+        For competing risks, the event of interest.
+
+    ipcw_estimator : {None or 'km'}, default="km"
+        The inverse probability of censoring weighted (IPCW) estimator.
+        - None set uniform weights to all samples
+        - "km" use the Kaplan-Meier estimator
+
+    tied_tol : float, default=1e-8
+        The tolerance range to consider two probabilities equal.
+
+    Returns
+    -------
+    cindex: (n_taus,)
+        Value of the concordance index for each tau in taus.
+
+    References
+    ----------
+    .. [1] Wolbers, M., Blanche, P., Koller, M. T., Witteman, J. C., &
+           Gerds, T. A. (2014).
+           Concordance for prognostic models with competing risks.
+           Biostatistics, 15(3), 526-539.
+
+    .. [2] Uno, H., Cai, T., Pencina, M. J., D'Agostino, R. B., &
+           Wei, L. J. (2011).
+           On the C‐statistics for evaluating overall adequacy of risk
+           prediction procedures with censored survival data.
+           Statistics in medicine, 30(10), 1105-1117.
+
+    .. [3] Gerds, T. A., Kattan, M. W., Schumacher, M., & Yu, C. (2013).
+           Estimating a time‐dependent concordance index for survival
+           prediction models with covariate dependent censoring.
+           Statistics in medicine, 32(13), 2173-2184.
+
+    .. [4] Blanche, P., Kattan, M. W., & Gerds, T. A. (2019).
+           The c-index is not proper for the evaluation of-year predicted risks.
+           Biostatistics, 20(2), 347-357.
+
+    """
+    c_index_report = _concordance_index_incidence_report(
+        y_test,
+        y_pred,
+        time_grid=time_grid,
+        taus=taus,
+        y_train=y_train,
+        X_train=X_train,
+        X_test=X_test,
+        event_of_interest=event_of_interest,
+        ipcw_estimator=ipcw_estimator,
+        tied_tol=tied_tol,
+    )
+    return c_index_report["cindex"]
+
+
+def _concordance_index_incidence_report(
+    y_test,
+    y_pred,
+    time_grid=None,
+    taus=None,
+    y_train=None,
+    X_train=None,
+    X_test=None,
+    event_of_interest=1,
+    ipcw_estimator="km",
+    tied_tol=1e-8,
+):
+    """Report version of function `concordance_index_incidence`.
+
+    Running this function directly is useful to get more insights about
+    the underlying statistics of the c-index.
+    All returned lists have the same length as taus.
+
+    Returns
+    -------
+    cindex: list of float
+        Value of the concordance index
+
+    n_pairs_a: list of int
+        Number of comparable pairs with T_i <= T_j (type A) without ties for D_j != 0
+        those are the only comparable pairs for survival without competing events.
+
+    n_concordant_pairs_a: list of int
+        Number of concordant pairs among A pairs without ties.
+
+    n_ties_times_a: list of int
+        Number of tied pairs of type A with D_i = D_j = 1.
```

Comment:
So here I believe the docstring is accurate, we consider ties of type A only individuals having experienced the event of interest simultaneously.
If you think it would bring clarity, we could write
```
D_i = D_j = 1 and T_j = T_i
```